File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> words = {};
        string result = "";

        for (int i = 0; i < s.length(); i++){
            string word = "";

            while (i < s.length() && s[i] != ' '){
                word += s[i];
            }

            if (word.length() > 0) words.push_back(word);
        }

        for (int i = words.size() - 1; i >= 0; i--){
            result += words[i];
            if (i != 0) result += " ";
        }

        return result;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        List<String> words = new ArrayList<>();
        String result = "";
        
        for (int i = 0; i < s.length(); i++){
            String word = "";

            while (i < s.length() && s.charAt(i) != ' '){
                word += s.charAt(i);
                i++;
            }

            if (word.length() > 0) words.add(word);
        }

        for (int i = words.size() - 1; i >= 0; i--){
            result += words.get(i);
            if (i != 0) 
                result += " ";
        }

        return result;
    }

}